sdk/search/azure-search-documents/azure/search/documents/_service/_search_service_client.py

```diff
@@ -59,6 +71,163 @@ def get_service_statistics(self, **kwargs):
+
+        :return: List of indexes
+        :rtype: list[~azure.search.documents.Index]
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        result = self._client.indexes.list(**kwargs)
+        return result.indexes
+
+    @distributed_trace
+    def get_index(self, index_name, **kwargs):
+        # type: (str, **Any) -> Index
+        """
+
+        :param index_name: The name of the index to retrieve.
+        :type index_name: str
+        :return: Index object
+        :rtype: ~azure.search.documents.Index
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        result = self._client.indexes.get(index_name, **kwargs)
+        return result
+
+    @distributed_trace
+    def get_index_statistics(self, index_name, **kwargs):
+        # type: (str, **Any) -> dict
+        """Returns statistics for the given index, including a document count
+        and storage usage.
+
+        :param index_name: The name of the index to retrieve.
+        :type index_name: str
+        :return: Statistics for the given index, including a document count and storage usage.
+        :rtype: ~azure.search.documents.GetIndexStatisticsResult
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        result = self._client.indexes.get_statistics(index_name, **kwargs)
+        return result.as_dict()
+
+    @distributed_trace
+    def delete_index(self, index_name, **kwargs):
+        # type: (str, **Any) -> None
+        """Deletes a search index and all the documents it contains.
+
+        :param index_name: The name of the index to retrieve.
+        :type index_name: str
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        self._client.indexes.delete(index_name, **kwargs)
+
+    @distributed_trace
+    def create_index(self, index, **kwargs):
+        # type: (Index, **Any) -> Index
+        """
+
+        :param index: The index object.
+        :type index: ~azure.search.documents.Index
+        :return: The index created
+        :rtype: ~azure.search.documents.Index
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        result = self._client.indexes.create(index, **kwargs)
+        return result
+
+    @distributed_trace
+    def create_or_update_index(
+            self,
+            index_name,
+            index,
+            allow_index_downtime=None,
+            match_condition=MatchConditions.Unconditionally,
+            **kwargs
+    ):
+        # type: (str, Index, bool, MatchConditions, **Any) -> Index
+        """
```

Comment:
It's unclear why there is create_index and create_or_update_index, definitely deserves some documentation. Is create_index an alias to here? Does it mean "create_index" cannot udpate? etc.

Reply:
The feature crew decides to expose both though create_or_update with if-non-match may have same behavior of create.